pkg/kubelet/managed/managed.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package managed
+
+import (
+	"os"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+var (
+	pinnedManagementEnabled      bool
+	pinnedManagementFilename     = "/etc/kubernetes/workload-pinning"
+	WorkloadsLabel               = "workload.openshift.io/target=management"
+	ManagedCapacityLabel         = "workload.openshift.io/cores"
+	ContainerCpuAnnotationFormat = "io.openshift.workload.management.cpu/%v"
+)
+
+func init() {
+	readEnablementFile()
+}
+
+func readEnablementFile() {
+	if _, err := os.Stat(pinnedManagementFilename); err == nil {
+		pinnedManagementEnabled = true
+	}
+}
+
+func IsEnabled() bool {
+	return pinnedManagementEnabled
+}
+
+func IsPodManaged(pod *v1.Pod) bool {
+	if pod.ObjectMeta.Labels == nil {
+		return false
+	}
+	_, found := pod.ObjectMeta.Labels[WorkloadsLabel]
+	return found
+}
+
+const (
+	// These limits are defined in the kernel:
+	// https://github.com/torvalds/linux/blob/0bddd227f3dc55975e2b8dfa7fc6f959b062a2c7/kernel/sched/sched.h#L427-L428
+	MinShares = 2
+	MaxShares = 262144
+
+	SharesPerCPU  = 1024
+	MilliCPUToCPU = 1000
+)
+
+// MilliCPUToShares converts the milliCPU to CFS shares.
```

Comment:
You already have library for it, search for `MilliCPUToShares`

Reply:
The kubelet can't use it due to a circular dependency.